### ISSUE TITLE
[SideNavigation][a11y] contrast does not meet 3 to 1 ratio 

### ISSUE THREAD

**Issue by jenndiaz** [adobe/spectrum-css#2210]
## Description

The selected SideNav item color does not meet 3 to 1 contrast ratio with the adjacent background colors. 

## Steps to reproduce

1. Go to [the docs site for side nav](https://opensource.adobe.com/spectrum-css/sidenav.html#standard) 
2. Observe the selected standard variant, the background color of the selected variant compared to the background color of the adjacent SideNav items background colors 

## Expected behavior

The contrast between the non-selected and selected SideNav items MUST be at least 3:1.

## Screenshots

![Screenshot 2023-10-12 at 10 32 36 AM](https://github.com/adobe/spectrum-css/assets/63808889/96ab0be6-8f83-453a-acce-c26049bc8094)

## Additional context

This is a failure of WCAG Success Criteria: [1.4.11 Non-text Contrast (Level AA) – WCAG 2.1](https://www.w3.org/TR/WCAG21/#non-text-contrast)
WCAG Understanding document: https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast

**Comment by castastrophe** [comment]
Current failure on latest code: `insufficient color contrast of 2.01 (foreground color: #b1b1b1, background color: #f8f8f8, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1` for `.spectrum-SideNav > .is-disabled.spectrum-SideNav-item > .spectrum-SideNav-itemLink[href="#"] > .spectrum-SideNav-link-text`

https://opensource.adobe.com/spectrum-css/?path=/story/components-side-nav--default&globals=testingPreview:!true
